lib/scala/filewatcher/src/main/scala/org/enso/filewatcher/WatcherFactory.scala

```diff
@@ -0,0 +1,19 @@
+package org.enso.filewatcher
+
+import java.nio.file.Path
+
+trait WatcherFactory {
```

Comment:
I'd make the `WatcherFactory` an inner class of `Watcher` if I was designing the API in Java.